cdap-app-templates/cdap-program-report/src/main/scala/co/cask/cdap/report/ReportGenerationHelper.scala

```diff
@@ -136,20 +136,35 @@ object ReportGenerationHelper {
     // TODO: [CDAP-13291] improve how the number of partitions is configured
     resultDf.coalesce(1).write.option("timestampFormat", "yyyy/MM/dd HH:mm:ss ZZ").json(reportDir)
     val count = resultDf.count
-    // Write the total number of records in _SUCCESS file generated after successful report generation
+    // Create a _COUNT file and write the total number of report records in it
+    writeToFile(count.toString, Constants.LocationName.COUNT_FILE, reportIdDir)
```

Comment:
why is this needed?

Reply:
this is for providing the total number of report records in the response of reading the report details without the need of iterating through the whole report file